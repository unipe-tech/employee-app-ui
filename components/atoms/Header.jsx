import { AppBar } from "@react-native-material/core";
import { TouchableOpacity } from "react-native";
import { MaterialIcons } from "react-native-vector-icons";
import { COLORS, FONTS } from "../../constants/Theme";

const Header = ({ title, onLeftIconPress, onRightIconPress }) => {
  return (
    <AppBar
      title={title}
      contentContainerStyle={{ height: 50 }}
      color={COLORS.primary}
      titleStyle={{ color: COLORS.white, ...FONTS.h3 }}
      leading={
        <TouchableOpacity>
          <MaterialIcons
            name="arrow-back"
            onPress={onLeftIconPress}
            size={20}
            color="white"
          />
        </TouchableOpacity>
      }
      trailing={
        onRightIconPress && (
          <TouchableOpacity>
            <MaterialIcons
              name="arrow-forward"
              onPress={onRightIconPress}
              size={20}
              color="white"
            />
          </TouchableOpacity>
        )
      }
    />
  );
};

export default Header;
