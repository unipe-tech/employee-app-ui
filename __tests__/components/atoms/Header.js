import Header from "../../../components/atoms/Header";
import { render, fireEvent, waitFor } from "@testing-library/react-native";
import { describe, expect, test } from "@jest/globals";

describe("Header", () => {
  test("renders UI correctly", () => {
    const tree = render(
      <Header
        title={"Head"}
        onLeftIconPress={() => jest.fn()}
        onRightIconPress={() => jest.fn()}
      />
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
