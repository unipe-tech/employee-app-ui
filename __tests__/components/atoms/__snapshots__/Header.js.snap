// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header renders UI correctly 1`] = `
<View
  collapsable={false}
  style={
    Object {
      "backgroundColor": "#2CB77C",
      "shadowColor": "#000",
      "shadowOffset": Object {
        "height": 2,
        "width": 0,
      },
      "shadowOpacity": 0.23,
      "shadowRadius": 2.62,
      "zIndex": 4,
    }
  }
>
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
          "flexDirection": "row",
          "height": 44,
          "justifyContent": "space-between",
          "paddingHorizontal": 16,
        },
        Object {
          "height": 50,
        },
      ]
    }
  >
    <View
      style={
        Array [
          Object {
            "marginEnd": 20,
            "marginStart": -12,
          },
          undefined,
        ]
      }
    >
      <View
        accessible={true}
        collapsable={false}
        focusable={false}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "opacity": 1,
          }
        }
      >
        <Text />
      </View>
    </View>
    <View
      style={
        Array [
          Object {
            "flex": 1,
            "justifyContent": "center",
          },
          undefined,
        ]
      }
    >
      <Text
        style={
          Array [
            Object {
              "fontSize": 20,
              "fontWeight": "500",
              "letterSpacing": 0.15,
            },
            Object {
              "color": "#000000",
            },
            Array [
              Object {
                "color": "#000000",
              },
              Object {
                "color": "#FFFFFF",
                "fontFamily": "Montserrat-Bold",
                "fontSize": 18,
                "lineHeight": 22,
              },
            ],
          ]
        }
      >
        Head
      </Text>
    </View>
    <View
      style={
        Array [
          Object {
            "marginEnd": -12,
            "marginStart": 20,
          },
          undefined,
        ]
      }
    >
      <View
        accessible={true}
        collapsable={false}
        focusable={false}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "opacity": 1,
          }
        }
      >
        <Text />
      </View>
    </View>
  </View>
</View>
`;
